### PR TITLE
added error boundary

### LINKED ISSUE #526: Show error page on app-crash
**Which part of the project should be enhanced?**
Probably the whole React-App-Wrapper.

**Is your enhancement request related to a problem? Please describe.**
When being in production build and an un-catchable error occurs, the screen will erase just to a blank gray page. This can be very surprising for a user.

**Describe the solution you'd like**
In case of an error we should show a page with some information that an error occurred and how to report it to us and a button to easily reload the app.

**Describe alternatives you've considered**
Automatically reload the page. But this might lead to even more confusion if the editor sometimes reloads by itself without further information.

**Additional context**
The react error boundaries seem to be useful for this: https://reactjs.org/docs/error-boundaries.html

### PR DESCRIPTION
**Pull request by DerMolly**
### Component/Part
the whole app

### Description
This PR adds an error boundary

### Steps

- [x] added implementation
- [ ] added / updated tests
- [ ] added / updated documentation
- [ ] extended changelog

### Related Issue(s)
fixes #526

### CODE REVIEW

File: src/components/error-boundary/error-boundary.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    console.error('error caught', error)
+    console.error('additional information', errorInfo)
+  }
+
+  render (): ReactElement | undefined | null | string | number | boolean {
+    if (this.state.hasError) {
+      // You can render any custom fallback UI
+      return (<LandingLayout>
+        <div className='text-white d-flex align-items-center justify-content-center my-5'>
+          <h1>An error occurred</h1>
```
Comment (ErikMichelson) [inline src/components/error-boundary/error-boundary.tsx:25]: I would like to see the following here:
 - a link to the issue tracker for reporting bugs (link can be obtained from version.json)
 - a button to reload the app

**Comment by DerMolly** [comment]
image

**Comment by DerMolly** [comment]
Translations seem to be a problem, but maybe the error handler can live without them?

**Comment by ErikMichelson** [comment]
Is it a good idea to include the landing layout in the error page?

Think of the situation that the UserAvatar component yields an error as example. The landing-layout uses the nav-bar which in turn uses the UserAvatar. That results in an error loop.

**Comment by DerMolly** [comment]
Hm, not a bad point I just wanted a non blank page, but I'll copy what's needed for that instead.

**Comment by DerMolly** [comment]
image

**Comment by mrdrogdrog** [comment]
POV: You're a user, the app crashes and you see this. Confusion is over 9000.

Conclusion: A bit more text would be nice. 
```
An unknown error occured.
Don't worry, this happens sometimes. If this is the first time you see this page then try reloading the app. 
If you can reproduce this error, then we would be glad if you open an issue on github or contact us on matrix.
```

**Comment by DerMolly** [comment]
image

**Review by ErikMichelson — APPROVED** [review]

**Comment by DerMolly** [comment]
image

**Review by ErikMichelson — APPROVED** [review]

**Review by mrdrogdrog — APPROVED** [review]

**Pull request merged**